force-app/main/default/lwc/propertyListMap/__tests__/propertyListMap.test.js

```diff
@@ -0,0 +1,189 @@
+        };
+    },
+    { virtual: true }
+);
+
+const MOCK_PROPERTIES = {
+    records: [
+        { Id: 'id1', Location__Latitude__s: 10, Location__Longitude__s: 11 },
+        { Id: 'id2', Location__Latitude__s: 20, Location__Longitude__s: 21 }
+    ]
+};
+
+// Sample error for loadScript error
+const LOAD_SCRIPT_ERROR = {
+    body: { message: 'An internal server error has occurred' },
+    ok: false,
+    status: 400,
+    statusText: 'Bad Request'
+};
+
+describe('c-property-list-map', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        // Clear mocks so that every test run has a clean implementation
+        jest.clearAllMocks();
+        // Clear leaflet global
+        global.L = undefined;
+    });
+
+    // Helper function to wait until the microtask queue is empty.
+    // Used when having to wait for asynchronous DOM updates.
+    async function flushPromises() {
+        return Promise.resolve();
+    }
+
+    it('registers propertyFilters subscriber during the component lifecycle', () => {
+        // Create component
+        const element = createElement('c-property-list-map', {
+            is: PropertyListMap
+        });
+        document.body.appendChild(element);
+
+        // Validate if subscriber got registered after connected to the DOM
+        expect(subscribe).toHaveBeenCalled();
+        expect(subscribe.mock.calls[0][1]).toBe(FILTERS_CHANGED);
+    });
+
+    it('loads the leaflet javascript and css static resources', () => {
+        // Create component
+        const element = createElement('c-property-list-map', {
+            is: PropertyListMap
+        });
+        document.body.appendChild(element);
+
+        // Validation that the loadScript and loadStyle promises
+        // are called once.
+        expect(loadScript.mock.calls.length).toBe(1);
+        expect(loadStyle.mock.calls.length).toBe(1);
+
+        // Validation that the JS and CSS files are passed as parameters.
+        expect(loadScript.mock.calls[0][1]).toEqual('leafletjs/leaflet.js');
+        expect(loadStyle.mock.calls[0][1]).toEqual('leafletjs/leaflet.css');
+    });
+
+    it('fires a toast event if the static resource cannot be loaded', async () => {
+        loadScript.mockRejectedValue(LOAD_SCRIPT_ERROR);
+
+        // Create component
+        const element = createElement('c-property-list-map', {
+            is: PropertyListMap
+        });
+        document.body.appendChild(element);
+
+        // Mock handler for toast event
+        const handler = jest.fn();
+        // Add event listener to catch toast event
+        element.addEventListener(ShowToastEventName, handler);
+
+        // Wait for any asynchronous DOM updates
+        await flushPromises();
+        await flushPromises();
```

Comment:
This is done twice!

Reply:
Yes, it's needed twice. I'll add some comments on it.

Reply:
Added some comments to explain why we need to wait twice here.